### PR TITLE
Update dependency go modules for k8s v1.27.0-rc.0

### PR DESCRIPTION
**Pull request by sunnylovestiramisu**
Ran kubernetes-csi/csi-release-tools go-get-kubernetes.sh -p 1.27.0-rc.0

```release-note
NONE
```

### CODE REVIEW

**Comment by sunnylovestiramisu** [comment]
Somehow the vulnerability scan job was cancelled

File: go.mod [hunk 1]
```diff
@@ -65,65 +66,65 @@ require (
...
 replace github.com/kubernetes-csi/external-snapshotter/client/v6 => ./client
 
-replace k8s.io/api => k8s.io/api v0.26.0
+replace k8s.io/api => k8s.io/api v0.27.0-rc.0
 
-replace k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.26.0
+replace k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.27.0-rc.0
 
-replace k8s.io/apimachinery => k8s.io/apimachinery v0.26.0
+replace k8s.io/apimachinery => k8s.io/apimachinery v0.28.0-alpha.0
```
Comment (xing-yang) [inline go.mod:82]: Why is this one v0.28.0-alpha.0, not v0.27.0-rc.0?

File: go.mod [hunk 2]
```diff
@@ -65,65 +66,65 @@ require (
...
+replace k8s.io/component-base => k8s.io/component-base v0.27.0-rc.0
 
-replace k8s.io/component-helpers => k8s.io/component-helpers v0.26.0
+replace k8s.io/component-helpers => k8s.io/component-helpers v0.27.0-rc.0
 
-replace k8s.io/controller-manager => k8s.io/controller-manager v0.26.0
+replace k8s.io/controller-manager => k8s.io/controller-manager v0.27.0-rc.0
 
-replace k8s.io/cri-api => k8s.io/cri-api v0.26.0
+replace k8s.io/cri-api => k8s.io/cri-api v0.28.0-alpha.0
```
Comment (xing-yang) [inline go.mod:102]: This one too.

**Pull request merged**